CRM/Contact/BAO/GroupContactCache.php

```diff
@@ -336,49 +309,44 @@ public static function remove($groupID = NULL, $onceOnly = TRUE) {
 DELETE     gc
 FROM       civicrm_group_contact_cache gc
 INNER JOIN civicrm_group g ON g.id = gc.group_id
-WHERE      g.cache_date <= %1
+WHERE      g.cache_date <= '$expiryTime'
 ";
         $update = "
 UPDATE civicrm_group g
-SET    cache_date = null,
-       refresh_date = null
-WHERE  g.cache_date <= %1
-";
-        $refresh = "
-UPDATE civicrm_group g
-SET    refresh_date = $refreshTime
-WHERE  g.cache_date > %1
-AND    refresh_date IS NULL
+SET    cache_date = null
+WHERE  g.cache_date <= '$expiryTime'
 ";
-        $cacheTime = date('Y-m-d H-i-s', strtotime("- $smartGroupCacheTimeout minutes"));
-        $params = array(1 => array($cacheTime, 'String'));
       }
     }
     elseif (is_array($groupID)) {
       $groupIDs = implode(', ', $groupID);
       $query = "
-DELETE     g
-FROM       civicrm_group_contact_cache g
-WHERE      g.group_id IN ( $groupIDs )
+DELETE     gc
+FROM       civicrm_group_contact_cache gc
+INNER JOIN civicrm_group g ON g.id = gc.group_id
+WHERE      g.id IN ( $groupIDs )
+AND        g.cache_date <= '$expiryTime'
 ";
       $update = "
 UPDATE civicrm_group g
-SET    cache_date = null,
-       refresh_date = null
+SET    cache_date = null
 WHERE  id IN ( $groupIDs )
+AND    g.cache_date <= '$expiryTime';
 ";
     }
     else {
```

Comment:
Since all these queries are identical, apart from the 'WHERE's, lets streamline them and just set the 'WHERE' differently depending on what's passed in?